Resources/views/Block/block_base.html.twig

```diff
@@ -12,21 +12,21 @@ file that was distributed with this source code.
 {% if sonata_page.isInlineEditionOn %}
     <div id="cms-block-{{ block.id }}"
          class="cms-block{% if block.hasParent() %} cms-block-element{% endif %}{% block block_class %}{% endblock %}"
-        {% if sonata_page.isEditor and block.page is defined %}
+        {% if sonata_page.isEditor and block.page is not null %}
             data-id="{{ block.id }}"
             data-name="{{ block.name }}"
             data-role="{% block block_role %}block{% endblock %}"
             data-page-id="{{ block.page.id }}"
         {% endif %}
         >
 {% elseif sonata_page.isEditor() %}
-    <!-- start rendering, block.id: {{ block.id }} - page.id: {% if block.page is defined %}{{ block.page.id }}{% else %}no related page{% endif %} - name: {{ block.name }} -->
+    <!-- start rendering, block.id: {{ block.id }} - page.id: {% if block.page|default(false) %}{{ block.page.id }}{% else %}no related page{% endif %} - name: {{ block.name }} -->
 {% endif %}
 
 {% block block %}EMPTY CONTENT{% endblock %}
 
 {% if sonata_page.isInlineEditionOn %}
     </div>
 {% elseif sonata_page.isEditor() %}
-    <!-- end rendering, block.id: {{ block.id }} - page.id: {% if block.page is defined %}{{ block.page.id }}{% else %}no related page{% endif %}  - name: {{ block.name }} -->
-{% endif %}
\ No newline at end of file
+    <!-- end rendering, block.id: {{ block.id }} - page.id: {% if block.page|default(false) %}{{ block.page.id }}{% else %}no related page{% endif %}  - name: {{ block.name }} -->
```

Comment:
Block/block_base.html.twig is used as base template also in the admin. In this case, the page variable is undefined.